ping.c

```diff
@@ -208,6 +209,8 @@ main(int argc, char **argv)
 #ifdef USE_IDN
 	setlocale(LC_ALL, "");
 #endif
+	if (!strcmp(setlocale(LC_ALL, NULL), "C"))
```

Comment:
If we use gcc with `-Wall`, then the lack of curly braces around the if statement body generates a warning.
```
root@ctrl:/src# gcc --version
gcc (Debian 7.3.0-19) 7.3.0
...
root@ctrl:/src# make ping
...
cc -O3 -g -fno-strict-aliasing -Wstrict-prototypes -Wall -D_GNU_SOURCE  -c ping.c -DCAPABILITIES -DUSE_IDN -DUSE_NETTLE  -o ping.o
ping.c: In function 'main':
ping.c:212:5: warning: this 'if' clause does not guard... [-Wmisleading-indentation]
     if (!strcmp(setlocale(LC_ALL, NULL), "C"))
     ^~
```

Reply:
Strange, I don't get this warning, I use the same environment. And there are other parts of `ping.c` with `if` without curly braces...
```
$ rm -f ping*.o ping; make
cc -O3 -g -fno-strict-aliasing -Wstrict-prototypes -Wall -D_GNU_SOURCE  -c ping.c -DCAPABILITIES -DUSE_IDN -DUSE_NETTLE  -o ping.o
cc -O3 -g -fno-strict-aliasing -Wstrict-prototypes -Wall -D_GNU_SOURCE  -c ping_common.c -DCAPABILITIES -DUSE_IDN -DUSE_NETTLE  -o ping_common.o
cc -O3 -g -fno-strict-aliasing -Wstrict-prototypes -Wall -D_GNU_SOURCE  -c ping6_common.c -DCAPABILITIES -DUSE_IDN -DUSE_NETTLE  -o ping6_common.o
cc   -O3 -g -fno-strict-aliasing -Wstrict-prototypes -Wall ping.o ping_common.o ping6_common.o -lcap -lidn2 -lnettle -lresolv -lm   -o ping
$ gcc --version
gcc (Debian 7.3.0-19) 7.3.0
```
And it's not either in any build log of this branch:
https://travis-ci.org/iputils/iputils/builds/383118273

Reply:
Yeah, never mind. PEBKAC. I can confirm that your change addresses the issue and doesn't introduce any compiler warnings.